### PR TITLE
implemented a workaround for Conul service discovary support. 

### PR DESCRIPTION
**Pull request by raga70**
implemented a workaround for Conul service discovery. Although the solution in this PR works  (please take it as  just guidance on how to fix it) , I did have time to get fully familiar with your code base and just aplied a hot fix.

PS:   my ide made a bit of a  reformatting on the files I touched (sorry for the extra hassle when reviewing ), if you have any questions send me a message

### CODE REVIEW

File: src/MMLib.SwaggerForOcelot/Repositories/DownstreamSwaggerDocsRepository.cs [hunk 1]
```diff
@@ -46,6 +46,12 @@ public class DownstreamSwaggerDocsRepository : IDownstreamSwaggerDocsRepository
             var clientName = _options.Value.HttpClientName ?? ((route?.DangerousAcceptAnyServerCertificateValidator ?? false) ? ServiceCollectionExtensions.IgnoreSslCertificate : string.Empty);
             var httpClient = _httpClientFactory.CreateClient(clientName);
 
+            if (!(url.StartsWith("http://") || url.StartsWith("https://"))) //the url that gets constructed when using consul misses the http schema , and trows an exception
```
Comment (Burgyn) [inline src/MMLib.SwaggerForOcelot/Repositories/DownstreamSwaggerDocsRepository.cs:49]: Use `StringComparison.InvariantCultureIgnoreCase`. and remove comments

**Comment by Burgyn** [comment]
Hi @raga70,

thanks for your PR. I can merge this PR if you:
- fix codefactor bot issues
- my comment
- and revert formatting of touched methods

thanks.

**Review by Burgyn — CHANGES_REQUESTED** [review]

**Comment by raga70** [comment]
Hi @Burgyn  I committed the changes you requested .

**Review by Burgyn — APPROVED** [review]

**Comment by Burgyn** [comment]
Hi @raga70,

unfortunately your changes failed tests.

Please fix tests.

**Comment by raga70** [comment]
I added nullability to my ServiceProviderType since the ServiceProviderConfiguration is in fact  null when unit testing

**Review by Burgyn — APPROVED** [review]

**Pull request merged**